src/libxively/tls/xi_tls_layer.c

```diff
@@ -333,6 +326,13 @@ static xi_state_t recv_handler( void* context, void* data, xi_state_t in_out_sta
     xi_data_desc_t* ret_buffer = layer_data->decoded_buffer;
     layer_data->decoded_buffer = NULL;
 
+    if ( NULL != layer_data->raw_buffer ||
+         xi_bsp_tls_pending( layer_data->tls_context ) > 0 )
```

Comment:
Whenever the data is left within the TLS read buffer and it does not reside in any external ( to the TLS's library buffer ) read_buffer.